src/CoreBundle/private/library/classes/TCMSTableEditor/TCMSTableEditorEndPoint.class.php

```diff
@@ -2096,6 +2090,44 @@ public function DeleteIdConnectedRecordReferences()
         }
     }
 
+    protected function deleteMultiTableRecordReferences(string $tableName, string $id = ''): void
+    {
+        $databaseConnection = $this->getDatabaseConnection();
+
+        $query = 'SELECT `id` FROM `cms_field_type` WHERE `constname` = '.$databaseConnection->quote(TCMSFieldExtendedLookupMultiTable::FIELD_SYSTEM_NAME);
+        $fieldTypeId = $databaseConnection->fetchColumn($query);
+
+        if (false === $fieldTypeId) {
+            return;
+        }
+
+        $query = 'SELECT 
+                         `cms_field_conf`.`name` AS fieldName,
+                         `cms_tbl_conf`.`name` AS tableName
+                    FROM `cms_field_conf` 
+               LEFT JOIN `cms_tbl_conf` ON `cms_tbl_conf`.`id` = `cms_field_conf`.`cms_tbl_conf_id` 
+                   WHERE `cms_field_conf`.`cms_field_type_id` = :fieldTypeId';
+        $result = $databaseConnection->fetchAll($query, ['fieldTypeId' => $fieldTypeId]);
+        foreach ($result as $row) {
+            $query = 'UPDATE 
+                         '.$databaseConnection->quoteIdentifier($row['tableName']).'
+                     SET 
+                         '.$databaseConnection->quoteIdentifier($row['fieldName'].TCMSFieldExtendedLookupMultiTable::TABLE_NAME_FIELD_SUFFIX)." = '',
+                         ".$databaseConnection->quoteIdentifier($row['fieldName'])." = ''   
+                   WHERE 
+                         ".$databaseConnection->quoteIdentifier($row['fieldName'].TCMSFieldExtendedLookupMultiTable::TABLE_NAME_FIELD_SUFFIX).' = '.$databaseConnection->quote($tableName);
+
+            if ('' !== $id) {
+                $query .= ' AND '.$databaseConnection->quoteIdentifier($row['fieldName']).' = '.$databaseConnection->quote($id);
+            }
+
+            $databaseConnection->query($query);
```

Comment:
How many entries/updates can this be?

Reply:
This depends on the usage of this fieldtype and the related tables. The query is a per table update, but only updates the records, that have the table name or table name and id set, so i would not call an update per record. This would be slower i'm sure.